homeassistant/components/volvooncall/device_tracker.py

```diff
@@ -1,45 +1,77 @@
 """Support for tracking a Volvo."""
 from __future__ import annotations
 
-from homeassistant.components.device_tracker import AsyncSeeCallback, SourceType
+from homeassistant import config_entries
+from homeassistant.components.device_tracker import SourceType
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.dispatcher import async_dispatcher_connect
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-from homeassistant.util import slugify
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from . import DATA_KEY, SIGNAL_STATE_UPDATED, VolvoUpdateCoordinator
+from . import DOMAIN, VolvoEntity, VolvoUpdateCoordinator
 
 
-async def async_setup_scanner(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    async_see: AsyncSeeCallback,
-    discovery_info: DiscoveryInfoType | None = None,
-) -> bool:
-    """Set up the Volvo tracker."""
-    if discovery_info is None:
-        return False
-
-    vin, component, attr, slug_attr = discovery_info
-    coordinator: VolvoUpdateCoordinator = hass.data[DATA_KEY]
-    volvo_data = coordinator.volvo_data
-    instrument = volvo_data.instrument(vin, component, attr, slug_attr)
-
-    if instrument is None:
-        return False
-
-    async def see_vehicle() -> None:
-        """Handle the reporting of the vehicle position."""
-        host_name = instrument.vehicle_name
-        dev_id = f"volvo_{slugify(host_name)}"
-        await async_see(
-            dev_id=dev_id,
-            host_name=host_name,
-            source_type=SourceType.GPS,
-            gps=instrument.state,
-            icon="mdi:car",
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Configure device_trackers from a config entry created in the integrations UI."""
+    volvo_data = hass.data[DOMAIN][config_entry.entry_id].volvo_data
+    for instrument in volvo_data.instruments:
+        if instrument.component == "device_tracker":
+            discovery_info = (
+                instrument.vehicle.vin,
+                instrument.component,
+                instrument.attr,
+                instrument.slug_attr,
+            )
+
```

Comment:
We don't need to create this tuple. Just pass the arguments explicitly in the call below.
```suggestion
```